src/creature.cpp

```diff
@@ -804,13 +815,60 @@ void Creature::onFollowCreature(const Creature*)
 		listWalkDir.clear();
 		onWalkAborted();
 	}
+}
 
-	hasFollowPath = false;
-	forceUpdateFollowPath = false;
-	isUpdatingPath = true;
+void Creature::onUnfollowCreature() { hasFollowPath = false; }
+
+// Pathfinding Events
+bool Creature::isFollower(Creature* creature)
+{
+	auto it = std::find(followers.begin(), followers.end(), creature);
+	if (it != followers.end()) {
+		return true;
+	}
+	return false;
+}
+
+void Creature::addFollower(Creature* creature)
+{
+	if (!isFollower(creature)) {
+		followers.push_back(creature);
+	}
 }
 
-void Creature::onUnfollowCreature() { isUpdatingPath = false; }
+void Creature::removeFollower(Creature* creature)
+{
+	auto it = std::find(followers.begin(), followers.end(), creature);
+	if (it != followers.end()) {
+		followers.erase(it);
+	}
+}
+
+void Creature::updateFollowersPaths()
+{
+	if (followers.empty()) {
+		return;
+	}
+
+	const Position& thisPosition = getPosition();
+	for (const auto follower : followers) {
```

Comment:
lets pass by reference in loops like this when we can please

const auto& follower